sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -1334,5 +1335,32 @@ public virtual async Task<Response<ReleaseKeyResult>> ReleaseKeyAsync(string nam
                 throw;
             }
         }
+
+        /// <summary>
+        /// Get a <see cref="CryptographyClient"/> for the given key.
+        /// </summary>
+        /// <param name="name">The name of the key used to perform cryptographic operations.</param>
+        /// <param name="version">Optional version of the key used to perform cryptographic operations.</param>
+        /// <returns>A <see cref="CryptographyClient"/> using the same options and pipeline as this <see cref="KeyClient"/>.</returns>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public virtual CryptographyClient GetCryptographyClient(string name, string version = null)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Uri keyUri = CreateKeyUri(VaultUri, name, version);
+            return new(keyUri, _pipeline);
```

Comment:
I'm not familiar with this syntax - how is it returning the CryptographyClient?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new

Roslyn recommends the change. While I'm a fan of `var`, @AlexGhiondea doesn't like them and I appreciate the reasoning, which fits in well with this `new()` pattern anyway: the lvalue is typed, and the rvalue/expression matters little.

Reply:
How does .NET re-uses the same _pipeline instead of creating a copy here?